src/Credentials/EcsCredentialProvider.php

```diff
@@ -54,11 +56,18 @@ public function __invoke()
     {
         $client = $this->client;
         $request = new Request('GET', self::getEcsUri());
+        
+        $authToken = self::getEcsAuthToken();
+        $headers = [];
+        if(!empty($authToken))
+            $headers = ['Authorization' => $authToken];
+        
         return $client(
             $request,
             [
                 'timeout' => $this->timeout,
                 'proxy' => '',
+                'headers' => $headers
```

Comment:
We want some component of a new test for the `EcsCredentialProvider` to ensure the Authorization header is set and the value matches the token

Reply:
Separated the $headers generate logic to separate public function in the same class to make it unit-testable. Added the code to the token. Will push the fix once we align.

Reply:
Code pushed.